core/include/moveit/task_constructor/cost_terms.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <moveit/task_constructor/storage.h>
+
+namespace moveit {
+namespace task_constructor {
+/// These structures all implement the Stage::CostTerm API and can be configured via Stage::setCostTerm()
+namespace cost {
+
+/// add a constant cost to each solution
+struct Constant
+{
+public:
+	Constant(double c) : cost(c) {}
+
+	double operator()(const SubTrajectory&) { return cost; }
+
+	double cost;
+};
+
+/// trajectory length (interpolated between waypoints)
+double PathLength(const SubTrajectory& s);
+
+/// execution duration of the whole trajectory
+double TrajectoryDuration(const SubTrajectory& s);
+
+struct LinkMotion
+{
+	LinkMotion(std::string link_name) : link_name(link_name) {}
+
+	double operator()(const SubTrajectory&, std::string&);
+
+	std::string link_name;
+};
+
+/** inverse distance to collision
+ *
+ * \arg interface compute distances using START or END interface of solution
+ * \arg with_world check distances to world objects as well or only look at self-collisions
+ * \arg cumulative if true, compute clearance as aggregated distance of all bodies
+ * \arg group_property the name of the property which defines the group to look at
```

Comment:
Hm. I would have expected that the average / min clearance is computed across the whole trajectory.
Maybe rename the class to StateClearance or StartEndClearance?